packages/x-license/src/utils/licenseErrorMessageUtils.ts

```diff
@@ -32,6 +32,16 @@ export function showLicenseKeyPlanMismatchError() {
   ]);
 }
 
+export function showProductScopeMismatchError() {
+  showError([
+    'MUI X: Invalid Product coverage.',
+    '',
+    'Your use of MUI X is not compatible with the plan of your license key. The product you are trying to use is not included in the version of your license plan. This happens if you try to use `ChartsPro` or TreeViewPro` with a license plan where these products are not included.',
```

Comment:
Yeah, not a fan of using `legacy`, it could be misleading to someone reading this error message in 2025 after some other pro-plan changes (is legacy pro plan the pro plan from 2023? the one from 2024?)

Maybe we could have a status specific to this plan change (and create another status when we do another pricing change) and that way we can have an error message that explains what is going on in details:

```ts
LICENSE_STATUS.NotAvailableInInitialProPlan
```

And the error could be something like:

```
The component you are trying to use is not included in your version of the Pro Plan.
You are using a license that is only compatible with the `@mui/x-data-grid-pro` and `@mui/x-date-pickers-pro` commercial packages. To start using another Pro package, please consider reaching to our sales team to upgrade your license.
```

Reply:
The exact error message would of course change based on https://github.com/mui/mui-x/pull/13459#discussion_r1644742468

Reply:
Sounds good!

Reply:
but we are not covering premium licenses with this wording.

Reply:
> but we are not covering premium licenses with this wording.

But we don't have any change on the premium plan do we?

Reply:
nope, but users with the premium plan will also run into this error when they use an "old" planversion

Reply:
It's probably nitpicking

Reply:
> I don't see a problem using 'legacy' 

The main issue is that for me "legacy" just means "an old/outdated version of something", when we will have 3, 4, 5 versions of the pro plan (which I'm pretty sure will happen), then `legacy` do not describe which version the person is using and which version he should be using.

At the very least I would rephrase to "an old version of the pro plan" to make this clear.
But I indeed prefer to be specific so that anyone in MUI can understand what is going on by reading the error message even in 3 years.

Reply:
I have added a continuation of this discussion in this PR #13568